tests/cpp/test_preseg_passes.cpp

```diff
@@ -642,4 +642,491 @@ TEST_F(PresegTest, ReplaceOutput) {
   testValidate(fec.fusion(), {out_tensor}, {in_tensor}, __LINE__, __FILE__);
 }
 
+namespace {
+void checkAmaxSegmentation(
+    FusionKernelRuntime& runtime,
+    int64_t number_of_segments,
+    int64_t fusion_index,
+    int64_t number_of_outputs_in_fusion,
+    int64_t number_of_iterdomains,
+    int64_t expected_number_of_reduction_axes) {
+  // Check number of fusion segments
+  std::vector<std::unique_ptr<Fusion>> segments = runtime.getFusionSegments();
+  EXPECT_EQ(segments.size(), number_of_segments);
+
+  // Check number of outputs in selected segment
+  Fusion* selected_fusion = segments.at(fusion_index).get();
```

Comment:
follow the previous nitpick, we are also only using the segments `Fusion` at fusion_index. So we don't really need to construct everything.